SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -46,31 +47,97 @@ UnitTest {
 		}
 	}
 
-	// run a single test in the name format "TestPolyPlayerPool:test_prepareChildrenToBundle"
-	*runTest { | methodName |
-		var class, method, unitTest;
-		# class, method = methodName.split($:);
-		class = class.asSymbol.asClass;
-		method.asSymbol;
-		method = class.findMethod(method.asSymbol);
-		if(method.isNil) { Error("Test method not found "+methodName).throw };
-		class.new.runTestMethod(method);
+	// run a class or single test in the name format
+	// "TestPolyPlayerPool:test_prepareChildrenToBundle"
+	*runTest { | testItem |
+		var method, failString;
+		case { testItem.isKindOf(Class) } {
+			if (testItem.isKindOf(Meta_UnitTest)) {
+				testItem.run(false, false)
+			} {
+				failString = "%: testItem % is not a subclass of UnitTest."
+			}
+		} { testItem.isKindOf(String) } {
+			method = this.findTestMethodFor(testItem);
+			if(method.notNil) {
+				method.ownerClass.new.runTestMethod(method)
+			} {
+				failString = "%: testItem % is not a valid testMethod string such as %."
+			}
+		} {
+			failString = "%: testItem % is neither a subclass of UnitTest,"
+			" nor a valid test-method string such as %."
+		};
+
+		if (failString.notNil) {
+			failString.format(thisMethod, testItem.cs, "TestABC:test_xyz".cs).warn
+		}
 	}
 
 	// run a single test method of this class
 	runTestMethod { | method |
-		var function;
+		var start = Main.elapsedTime, runTime;
 		("RUNNING UNIT TEST" + this.class.name ++ ":" ++ method.name).inform;
 		this.class.forkIfNeeded {
 			this.setUp;
 			currentMethod = method;
 			this.perform(method.name);
 			this.tearDown;
-			this.class.report;
+			runTime = Main.elapsedTime - start;
+			this.class.report(runTime);
 			nil
 		}
 	}
 
+	// run a list of testItems which can be UnitTest classes or method names
+	// in the format "TestPolyPlayerPool:test_prepareChildrenToBundle"
+
+	*runTests { |testItems, postTimes = false|
+		^this.forkIfNeeded {
+			var start = Main.elapsedTime;
+			this.reset;
+			testItems = testItems ?? { this.allSubclasses };
+			testItems.do { |testItem|
+				UnitTest.runTest(testItem);
+				0.1.wait;
```

Comment:
why is this 0.1? this could potentially add a lot of time to testing. is there anything wrong with e.g. (1e-6).wait?

Reply:
I put the 0.1.wait in here for consistency with runAll:
```
	*runAll {
		^this.forkIfNeeded {
			this.reset;
			this.allSubclasses.do ({ |testClass|
				testClass.run(false,false);
				0.1.wait;
			});
			this.report;
		}
	}
```

Reply:
Do you know why that method has that wait? Still seems suspicious to me but we can handle it later.